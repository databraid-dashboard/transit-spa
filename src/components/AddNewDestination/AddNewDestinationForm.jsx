import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

class AddNewDestinationForm extends Component {
  state = { destination: '' };

  handleChange = (e, { destination, value }) => this.setState({ [destination]: value });

  handleSubmit = () => this.setState({ destination: '' });

  render() {
    const { destination } = this.state;

    return (
      <Form onSubmit={this.handleSubmit}>
        <Form.Group>
          <Form.Input
            placeholder="New destination"
            destination="destination"
            value={destination}
            onChange={this.handleChange}
          />
          <Form.Button content="Submit" color="green" onClick={this.props.onClick} />
        </Form.Group>
      </Form>
    );
  }
}

AddNewDestinationForm.propTypes = {
  onClick: PropTypes.func.isRequired,
};

AddNewDestinationForm.defaultProps = {
  onClick: () => {},
};

export default AddNewDestinationForm;
