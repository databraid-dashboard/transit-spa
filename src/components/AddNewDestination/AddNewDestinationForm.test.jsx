import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddNewDestinationForm from './AddNewDestinationForm';

describe('AddNewDestinationForm', () => {
  it('should render blank form', () => {
    const component = shallow(<AddNewDestinationForm />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
