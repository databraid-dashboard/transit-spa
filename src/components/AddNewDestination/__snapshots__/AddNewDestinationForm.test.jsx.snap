// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewDestinationForm should render blank form 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormGroup>
    <FormInput
      as={[Function]}
      control={[Function]}
      destination="destination"
      onChange={[Function]}
      placeholder="New destination"
      value=""
    />
    <FormButton
      as={[Function]}
      color="green"
      content="Submit"
      control={[Function]}
      onClick={[Function]}
    />
  </FormGroup>
</Form>
`;
